test/unit/test_connection_factory.py

```diff
@@ -177,6 +188,181 @@ def test_get_saas_kwargs(
+    for e_var in e_vars:
+        env = {} if e_var is None else {"name": e_var}
+        assert optional_bool_from_env(env, "name") == expected
+
+
+def test_optional_bool_from_env_error() -> None:
+    with pytest.raises(ValueError, match="boolean"):
+        assert optional_bool_from_env({"name": "maybe"}, "name")
+
+
+@dataclass
+class SslTestOpt:
+    env_name: str
+    ssl_name: str | None = None
+    env_value: str | None = None
+    ssl_value: Any | None = None
+    is_file: bool = False
+    is_dir: bool = False
+
+    def get_test_values(self, tmp_path: Path) -> tuple[str | None, Any | None]:
+        if not (self.is_file or self.is_dir):
+            return self.env_value, self.ssl_value
+        pth = tmp_path / self.env_value
+        if self.is_dir:
+            pth.mkdir()
+        else:
+            pth.write_text("content", encoding="utf-8")
+        str_pth = str(pth)
+        return str_pth, str_pth
+
+
+@pytest.mark.parametrize(
+    "test_opts",
+    [
+        [
+            SslTestOpt(
+                env_name=ENV_SSL_CERT_VALIDATION,
+                env_value="true",
+                ssl_name="cert_reqs",
+                ssl_value=ssl.CERT_REQUIRED,
+            )
+        ],
+        [
+            SslTestOpt(
+                env_name=ENV_SSL_CERT_VALIDATION,
+                env_value="false",
+                ssl_name="cert_reqs",
+                ssl_value=ssl.CERT_NONE,
+            )
+        ],
+        [SslTestOpt(env_name=ENV_SSL_CERT_VALIDATION, ssl_name="cert_reqs")],
+        [
+            SslTestOpt(
+                env_name=ENV_SSL_TRUSTED_CA,
+                env_value="my_ca_dir",
+                is_dir=True,
+                ssl_name="ca_cert_path",
+            )
+        ],
+        [
+            SslTestOpt(
+                env_name=ENV_SSL_TRUSTED_CA,
+                env_value="my_ca_file.cert",
+                is_file=True,
+                ssl_name="ca_certs",
+            )
+        ],
+        [
+            SslTestOpt(
+                env_name=ENV_SSL_CLIENT_CERT,
+                env_value="my_cli_file.cert",
+                is_file=True,
+                ssl_name="certfile",
+            )
+        ],
+        [
+            SslTestOpt(
+                env_name=ENV_SSL_CLIENT_CERT,
+                env_value="my_cli_file.cert",
+                is_file=True,
+                ssl_name="certfile",
+            ),
+            SslTestOpt(
+                env_name=ENV_SSL_PRIVATE_KEY,
+                env_value="my_private.key",
+                is_file=True,
+                ssl_name="keyfile",
+            ),
+        ],
+    ],
+    ids=[
+        "cert_reqs_required",
+        "cert_reqs_none",
+        "cert_reqs_unknown",
+        "ca_cert_path",
+        "ca_certs",
+        "certfile",
+        "keyfile",
+    ],
+)
+def test_get_ssl_options(test_opts, tmp_path) -> None:
+    env: dict[str, str] = {}
+    expected: dict[str, Any] = {}
+    for test_opt in test_opts:
```

Comment:
Do I understand this correctly that we merge the test ops in env and expected with this loop. Is there a reason why we need multiple test ops then?